File: TravellerCreator/ClientApp/src/app/sector-creator/models/mongoose/mongoose-star-system.ts
import {IStarSystem} from "../interfaces/star-system";
import {StarSystemType} from "../enums/star-system-type";
import {MongoosePlanet} from "./mongoose-planet";
import {IStar} from "../interfaces/star";

export class MongooseStarSystem implements IStarSystem {
  type: StarSystemType;
  gasGiant: boolean;
  planets: MongoosePlanet[];
  stars: IStar[];
}

File: TravellerCreatorModels/SectorCreator/Mongoose/MongooseStarSystem.cs
using TravellerCreatorGlobalMethods;
using TravellerCreatorModels.SectorCreator.Enums;
using TravellerCreatorModels.SectorCreator.Interfaces;

namespace TravellerCreatorModels.SectorCreator.Mongoose;

public class MongooseStarSystem : IStarSystem
{
    public StarSystemType Type { get; set; } = StarSystemType.Basic;
    public bool GasGiant { get; set; }
    public List<IPlanet> Planets { get; set; } = new();
    public List<IStar> Stars { get; set; } = new();

    public void GetGasGiant()
    {
        if (Roll.D6(2) > 5) {
            GasGiant = true;
        } else if (Roll.D6(2) < 5) {
            GasGiant = true;
        }
    }
}